manifest.go

```diff
@@ -324,6 +325,78 @@ func ValidateProjectRoots(c *Ctx, m *Manifest, sm gps.SourceManager) error {
 	return valErr
 }
 
+// ValidatePackageRules ensure that there are no ineffectual package declarations
+// in the constraints, overrides, required, and ignored rules in the manifest.
```

Comment:
I don't think we should do this with `required` and `ignored`, because they can be used to include and ignore packages that are not part of root project code or not directly reachable by the root project.

Reply:
I reread the spec for `required` and I agree it shouldn't/ be applied in this case. 

But for `ignored` I can see value in letting you know that one of the packages in that rule isn't being imported.

Reply:
Let me know where we'd like to go with this when you get a chance!